lib/src/widgets/yaru_page_indicator.dart

```diff
@@ -1,6 +1,11 @@
 import 'package:flutter/material.dart';
 
 import 'yaru_carousel.dart';
+import 'yaru_page_indicator_theme.dart';
+
+typedef DotDecorationBuilder = Decoration Function(
+  int index,
```

Comment:
I was asking this question to myself.
But yeah we definitively should, because the theme definition can be far away from the page indicator itself, and so we can't know the current index and the page length.